test/xbgpu/test_engine.py

```diff
@@ -1047,7 +1050,9 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
         verify_corrprod_data(
             corrprod_outputs=corrprod_outputs,
             corrprod_results=corrprod_results,
-            batch_index_tuples=batch_index_tuples,
+            # Use the batch indices that start *after* the first
+            # purposefully-empty accumulation
+            batch_indices=test_batch_indices[HEAPS_PER_FENGINE_PER_CHUNK:],
```

Comment:
Glad you suggested this. I had something vastly similar in a previous iteration of the test (maybe even the previous `HEAD`).

I think I was getting a bit iffy with the amount of things being returned and passed around (also because I'm on two self-imposed strikes with `mypy`'s "test_engine_end_to_end is too complex").

Still, will update accordingly now.